### ISSUE TITLE
Question: validating modules

### ISSUE THREAD

**Issue by wyardley** [terraform-linters/tflint#927]
Let's say I have the following directory structure / setup:
```
├── modules
│   └── test_gce
│       ├── main.tf
│       └── variables.tf
└── projects
    └── foo
        └── main.tf
```

in `modules/test_gce/main.tf`:
```hcl
resource "google_compute_instance" "foobar" {
  name         = var.name
  machine_type = var.machine_type
  zone         = var.location

  boot_disk {
    auto_delete = "true"
    device_name = "test"

    initialize_params {
      image = "ubuntu-os-cloud/ubuntu-2004-focal-v20200701"
      size  = 10
      type  = "pd-standard"
    }
  }

  network_interface {
    subnetwork = "default"
  }

}
```

in `variables.tf`:
```hcl
variable "name" {
  default = ""
  type    = string
}

variable "machine_type" {
  default = "f1-micro"
  type    = string
}

variable "location" {
  default = "us-central1=a"
  type    = string
}
```

in `projects/foo/main.tf`
```hcl
module "test_gce" {
  source       = "../../modules/test_gce"
  name         = "foobar"
  machine_type = "invalidtype"
}
```

While I know this specific resource type isn't working anyway, is it expected that this will work? I have tested resource types that do fail when defined resource-like, but not when running `tflint --module` (after a `terraform init`) with an equivalent configuration.

Let me know if I'm misunderstanding how it's intended to work, or if there's additional details I can provide.

**Comment by bendrucker** [comment]
Yes, this should work. The documented [example of module inspection](https://github.com/terraform-linters/tflint/blob/master/docs/guides/advanced.md#module-inspection) is virtually identical but deals with AWS.

**Comment by bendrucker** [comment]
Err, can you share your TFLint config? The Google plugin is literally hours old, and I'm not sure a rule covering instance types actually exists...

**Comment by wyardley** [comment]
yeah, instance types doesn't exist yet. I tested it with another resource type (bigtable instance names), but I'll come up with a better repro case for here with a resource that is supported and fails when the resource is defined resource-like

```hcl
plugin "google" {
    enabled = true
}
```

is all I have in my config.

**Comment by wyardley** [comment]
I thought I tested this before, but apparently not well enough. This test case works well for me:
in `modules/test_memorystore/main.tf`:

```hcl
resource "google_redis_instance" "default" {
  name               = var.name
  tier               = "STANDARD_HA"
  memory_size_gb     = 5
  display_name       = var.name
  region             = var.region
  redis_version      = "5_0"
  authorized_network = "default"
  redis_configs      = {}
}
```

in `projects/foo/main.tf`:
```hcl
module "test_memorystore" {
  source = "../../modules/test_memorystore"
  name   = "10$$$fake"
}

resource "google_redis_instance" "test" {
  name               = "10$$$fake"
  tier               = "STANDARD_HA"
  memory_size_gb     = 5
  display_name       = var.name
  region             = var.region
  redis_version      = "5_0"
  authorized_network = "default"
  redis_configs      = {}
}
```
```
% tflint --module
2 issue(s) found:

Error: "name" ("10$$$fake") doesn't match regexp "^[a-z][a-z0-9-]{0,39}[a-z0-9]$" (google_redis_instance_invalid_name)

  on main.tf line 3:
   3:   name   = "10$$$fake"

Callers:
   main.tf:3,12-23
   ../../modules/test_memorystore/main.tf:2,24-32

Error: "name" ("10$$$fake") doesn't match regexp "^[a-z][a-z0-9-]{0,39}[a-z0-9]$" (google_redis_instance_invalid_name)

  on main.tf line 7:
   7:   name               = "10$$$fake"

```
